### ISSUE TITLE
Define: Line length (and what to do when some long string forces us over it)

### ISSUE THREAD

**Issue by mtlynch** [m-lab/ansible-style-guide#3]

**Comment by pboothe** [comment]
I would be fine with the rule: prefer lines of less than 100 characters. But have that be a soft limit, so reviewers could decide on exceptions on a case-by-case basis. 

Sometimes filenames get long and there's no natural break, and I don't want that to run afoul of our rules.

**Comment by nkinkade** [comment]
I agree that a soft limit is probably better. There are only about 14 people left in the world that are still consistently developing on 80 column consoles.  Sometimes breaking long lines makes things much more clean and readable, but other times it's very awkward and forced and  makes this uglier than they otherwise would be.

**Comment by mtlynch** [comment]
Keep in mind some of our tools assume 80 columns. I believe Reviewable does this. I'd prefer a soft limit of 80.

**Comment by pboothe** [comment]
Soft limit of 80 is fine with me. The "soft limit" part is much more
important than 80 vs 100.

  -Peter

On Tue, May 17, 2016 at 3:02 PM, Michael Lynch notifications@github.com
wrote:

> Keep in mind some of our tools assume 80 columns. I believe Reviewable
> does this. I'd prefer a soft limit of 80.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> https://github.com/m-lab/ansible-style-guide/issues/3#issuecomment-219820072

## 

ᴹ̶LAB http://measurementlab.net/ | Measure the Internet, save the data,
and make it universally accessible and useful.

**Comment by stephen-soltesz** [comment]
Just for the record, reviewable makes it pretty easy to adjust the column width for all files in a PR. (See the upper right triangle on each file tab.)

What other tools require 80 columns for ansible?
